### PR TITLE
5955-Dont-add-events-if-they-are-too-far-in-the-future

### PR DESCRIPTION
**Pull request by markmckinnon**
Do not add timeline events if dates for files and artifacts are 12 years or greater from the current date/time.

### CODE REVIEW

File: bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java [hunk 1]
```diff
@@ -605,8 +619,11 @@ private long addEventDescription(long dataSourceObjId, long fileObjId, Long arti
 			return Optional.empty();
 		}
 		long time = eventPayload.getTime();
-		// if the time is legitimate ( greater than zero ) insert it into the db
-		if (time <= 0) {
+		// if the time is legitimate ( greater than zero and less then 12 years from present time) insert it into the db
+		if (time <= 0 || time >= MAX_TIMESTAMP_TO_ADD) {
+			if (time >= MAX_TIMESTAMP_TO_ADD) {
+				logger.log(Level.WARNING, String.format("Date/Time discarded from Timeline for %s for artifact %s", artifact.getDisplayName(), eventPayload.getDescription(TimelineLevelOfDetail.HIGH)));
```
Comment (raman-bt) [inline bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java]: Include the source file object id in the log.

File: bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java [hunk 2]
```diff
@@ -25,6 +25,7 @@
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
+import java.time.Instant;
```
Comment (raman-bt) [inline bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java]: Update the year in Copyright statement in file header to 2020 (and in all affected files)

File: bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java [hunk 3]
```diff
@@ -501,7 +511,7 @@ private long addEventDescription(long dataSourceObjId, long fileObjId, Long arti
 
 			for (Map.Entry<TimelineEventType, Long> timeEntry : timeMap.entrySet()) {
 				Long time = timeEntry.getValue();
-				if (time > 0) {// if the time is legitimate ( greater than zero ) insert it
+				if (time > 0 || time <= MAX_TIMESTAMP_TO_ADD) {// if the time is legitimate ( greater than zero or less then 12 years from current date) insert it
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java]: @markmckinnon and @raman-bt, shouldn't this be an AND test? With short-circuit evaluation of the conditional, all times greater than zero will pass the test.

Comment (raman-bt) [reply bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java]: You're right Richard. Ugh, I missed that.

**Comment by rcordovano** [comment]
@raman-bt, please re-review, thanks.

File: bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java (hunk 3 above)
Comment (markmckinnon) [reply bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java]: Fixed

File: bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java (hunk 2 above)
Comment (markmckinnon) [reply bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java]: FIxed

File: bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java (hunk 1 above)
Comment (markmckinnon) [reply bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java]: Fixed

**Pull request merged**